pyproject.toml

```diff
@@ -183,6 +183,35 @@ features = ['docs']
 build = "cd docs && make html"
 serve = "sphinx-autobuild docs docs/_build --host 0.0.0.0"
 
+[tool.hatch.envs.upstream]
+dependencies = [
+    'numpy>=2.2.0.dev0',
+    'numcodecs @ git+https://github.com/zarr-developers/numcodecs',
```

Comment:
It would be great if we could get `numcodecs` up on the scientific-python-nightly-wheels index...

https://github.com/zarr-developers/numcodecs/issues/313